### ISSUE TITLE
@nhost/react: Infinite loop with requests to `/token` if two tabs are open & user logs out in one tab

### ISSUE THREAD

**Issue by felixhagspiel** [nhost/nhost#2635]
**Describe the bug**
nhost react sdk starts inifinite number of calls loop to `/v1/token` if two tabs are open & user logs out in one tab & the auth token expires in the other tab.
This results in hundreds of requests to  `/v1/token` per minute returning `401`.

**To Reproduce**
1. setup a react project with versions:
```
  "@apollo/client": "^3.9.5",
    "@nhost/react": "^3.2.3",
    "@nhost/react-apollo": "^9.0.3"
```
2. start project
3. login
4. open the project in 2 tabs
5. logout in one tab, but do not reload the other tab
6. do not close the tabs, but wait ~15 minutes (TTL of JWT token) until the refresh token call in the tab in the background is started
7. it will start an infinite loop trying to get a new auth token

**Expected behavior**
It should not start an infinite loop, but remove the refresh token if its not existing / invalid 

**Screenshots**
<img width="601" alt="image" src="https://github.com/nhost/nhost/assets/5927172/8eebeb83-bebd-4c4a-a438-b1a7b54ccb04">

**Desktop (please complete the following information):**
 - OS: mac os sonoma
 - Browser: Chrome 123.0.6312.87 (arm64) 

**Additional context**
I assume the issue is that the refresh token is not removed from local storage when the server returns `401/refresh token not found`.

`_app.tsx`:
```
<NhostProvider nhost={nhostClient}>
  <NhostApolloProvider nhost={nhostClient} fetchPolicy="no-cache">
    {...other stuff}
  </NhostApolloProvider>
</NhostProvider>
```

**Comment by felixhagspiel** [comment]
any news on this?

**Comment by dbarrosop** [comment]
No, apologies. Other priorities got in the way. We will start looking into this today.

**Comment by felixhagspiel** [comment]
hey @onehassan the issue seems to be back. We are receiving a lot of requests:

```
{.... "message":"POST /v1/token 401 5ms","method":"POST","reason":{"error":"invalid-refresh-token","message":"Invalid or expired refresh token","status":401},"statusCode":401,"url":"/v1/token"}
```
<img width="142" alt="image" src="https://github.com/nhost/nhost/assets/5927172/141303a0-f539-4672-8fa2-a7e51ce37ed0">

**Comment by onehassan** [comment]
@felixhagspiel thanks for reporting, will investigate this issue further and keep you updated

**Comment by onehassan** [comment]
@felixhagspiel we've identified the issue and we have released a fix for this. Make sure to update to the latest versions of the SDK.


```
"@nhost/react": "^3.5.3",
"@nhost/react-apollo": "^12.0.3"
```

**Comment by felixhagspiel** [comment]
@onehassan looks like it still is an issue
<img width="226" alt="image" src="https://github.com/user-attachments/assets/8e86ddfa-4543-4c35-ae5c-920de8cf94ff">

**Comment by onehassan** [comment]
@felixhagspiel will double check — in the meantime can you please provide more details. Is it still reproducible as suggested in the description of this issue ?

**Comment by felixhagspiel** [comment]
@onehassan will check, I just saw that we are not at 5million requests, which should definetly not be the case :) then I checked the auth logs and saw thousands of entries. will try to reproduce it.

**Comment by dbarrosop** [comment]
as far as we know this issue was fixed, if you are still running into this issue using the latest versions of the SDK don't hesitate to let us know.
